### PR TITLE
Zero wallet integration

### PR DESCRIPTION
**Pull request by hasankhadra**
Added Zero Wallet integration.
Dockerized the app

### CODE REVIEW

File: src/routes/gasless-routes.ts [hunk 1]
```diff
@@ -38,57 +49,83 @@ interface ISendReq {
...
+
+
+/**
+ * Deploy the smart contract wallet
+ */
+// eslint-disable-next-line @typescript-eslint/require-await
+async function deploy(req: IReq<IDeployReq>, res: IRes) {
+  const { zeroWalletAddress, gasTankName, webHookAttributes } = req.body;
+
+  const newWebHookAttributes = {
```
Comment (hasankhadra) [inline src/routes/gasless-routes.ts:108]: Will fix this hack tomorrow

File: src/routes/gasless-routes.ts [hunk 2]
```diff
@@ -38,57 +49,83 @@ interface ISendReq {
...
+    ...webHookAttributes,
+    to: 'to_be_removed',
+    chainId: 0,
+  };
```
Comment (hasankhadra) [inline src/routes/gasless-routes.ts:112]: Will fix this hack tomorrow

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+COPY package*.json ./
+
+RUN npm install
+# for production
+# RUN npm ci --only=production
+
+COPY . .
+
+EXPOSE 3000
```
Comment (madhavanmalolan) [inline Dockerfile:13]: Can we please run this on 80 & 443 too?
The ssl keys should be a docker parameter

**Review by madhavanmalolan — CHANGES_REQUESTED** [review]

File: src/declarations/major/EnvVars.ts [hunk 4]
```diff
@@ -3,6 +3,11 @@
 export default {
   nodeEnv: (process.env.NODE_ENV ?? ''),
   port: (process.env.PORT ?? 0),
```
Comment (adiwajshing) [inline src/declarations/major/EnvVars.ts:5]: why do we have a duplicate declaration of PORT?

File: src/index.ts [hunk 5]
```diff
@@ -7,5 +7,10 @@ import server from './server';
 
 // **** Start server **** //
 
-const msg = ('Express server started on port: ' + EnvVars.port.toString());
-server.listen(EnvVars.port, () => logger.info(msg));
+const msg = (port: number | string) => (
+  'Express server started on port: ' + port.toString()
+);
+
+server.listen(EnvVars.ports[0], () => logger.info(msg(EnvVars.ports[0])));
```
Comment (adiwajshing) [inline src/index.ts:14]: why exactly are we starting the server on 3 ports?

File: example.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (adiwajshing) [inline example.yaml:1]: this should be called something more descriptive like "config.sample.yaml"

File: example.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+ databaseConfig:
+      user : "ENTER_USER_HERE"
+      host : "ENTER_HOST_HERE" # use 'host.docker.internal' in case of connecting to a local database
+      database: "ENTER_DB_NAME_HERE"
+      password  : "ENTER_PASSWORD_HERE"
+      port: "ENTER_PORT_WITHOUT_QUOTES_HERE"
```
Comment (adiwajshing) [inline example.yaml:7]: shouldn't matter if it's quotes or not, in the TS code -- it's easy to simply parse via `+port`

File: src/declarations/major/EnvVars.ts (hunk 4 above)
Comment (hasankhadra) [reply src/declarations/major/EnvVars.ts]: Fixed.

File: src/index.ts (hunk 5 above)
Comment (hasankhadra) [reply src/index.ts:14]: Check Madhavan comment above.

File: example.yaml (hunk 7 above)
Comment (hasankhadra) [reply example.yaml]: Fixed.

File: src/routes/gasless-routes.ts (hunk 2 above)
Comment (hasankhadra) [reply src/routes/gasless-routes.ts]: Fixed.

File: Dockerfile (hunk 3 above)
Comment (hasankhadra) [reply Dockerfile]: Fixed.

Comment (hasankhadra) [reply Dockerfile]: Fixed the first point. 
Let's discuss more on the ssl point.

File: env/development.env [hunk 8]
```diff
@@ -4,6 +4,8 @@ NODE_ENV=development
 
 ## Server ##
 PORT=3000
+PORT_2=80
```
Comment (adiwajshing) [inline env/development.env:7]: can remove

File: env/production.env [hunk 9]
```diff
@@ -4,6 +4,8 @@ NODE_ENV=production
 
 ## Server ##
 PORT=8081
+PORT_2=80
+PORT_3=443
```
[the remark this answers is not in the document]
Comment (adiwajshing) [reply env/production.env:8]: ideally, shouldn't commit the production env

File: package.json [hunk 10]
```diff
@@ -28,6 +28,7 @@
   },
   "dependencies": {
     "@ethersproject/abstract-provider": "^5.7.0",
+    "@hasankhadra/zero-wallet-server": "^1.0.3",
```
Comment (adiwajshing) [inline package.json:31]: probably best to rename the package to "@questbook/zero-wallet-server"

File: openapi.yaml [hunk 11]
```diff
@@ -0,0 +1,425 @@
...
+              $ref: '#/components/schemas/AuthorizeRequest'
+      responses:
+        '200':
+          $ref: '#/components/responses/AuthorizeResponse'
+        '400':
+          $ref: '#/components/responses/ErrorResponse'
+        '500':
+          $ref: '#/components/responses/ErrorResponse'
+
+  /auth/getNonce:
```
Comment (adiwajshing) [inline openapi.yaml:37]: best to stick to a single type of casing on routes, either snake or camel

File: package.json (hunk 10 above)
Comment (hasankhadra) [reply package.json:31]: I wasn't able to publish it with @questbook, I'll check with Madhavan what's needed to get this done

File: env/development.env (hunk 8 above)
Comment (hasankhadra) [reply env/development.env:7]: Fixed

File: openapi.yaml (hunk 11 above)
Comment (hasankhadra) [reply openapi.yaml:37]: Fixed.

File: env/production.env (hunk 9 above)
[the remark this answers is not in the document]
Comment (hasankhadra) [reply env/production.env]: Fixed.

**Pull request merged**